### PR TITLE
add EquFlash to Matbench Discovery leaderboard

### PR DESCRIPTION
**Pull request by choant2003**
## Description

   EquFlash is an E(3)-equivariant model based on the SevenNet-0 architecture, with tensor products accelerated by FlashTP(https://openreview.net/forum?id=wiQe95BPaB) developed in Materials AI Lab at Samsung Electronics.

The submission also includes:

- metadata, evaluation metrics, resources, and an interactive energy-parity chart
- batched structural relaxation with trajectory handling and FIRE or L-BFGS optimization
- batch FC3 thermal-conductivity evaluation and aggregation/evaluation CLI tools
- example runners and tests for discovery and conductivity workflows

## Checklist

Please check the following items before submitting your PR:

- [X] I created a new folder and YAML metadata file `models/<arch_name>/<model_variant>.yml` for my submission. `arch_name` is the name of the architecture and `model_variant.yml` includes things like author details, training set names and important hyperparameters.
- [X] I added the new model as a new attribute [`Model.<arch_name>` enum](https://github.com/janosh/matbench-discovery/blob/57d0d0c8a14cd317/matbench_discovery/enums.py#L274) on the `Model` enum in `enums.py`.
- [X] I uploaded the energy/force/stress model prediction file for the WBM test set to Figshare or another cloud storage service (`<yyyy-mm-dd>-<model_variant>-preds.csv.gz`).
- [X] I uploaded the model-relaxed structures file to Figshare or another cloud storage service in [JSON lines format](https://jsonlines.org) (`<yyyy-mm-dd>-wbm-IS2RE-FIRE.jsonl.gz`). JSON Lines allows fast loading of small numbers of structures with `pandas.read_json(lines=True, nrows=100)` for inspection.
- [X] I uploaded the phonon predictions to Figshare or another cloud storage service (`<yyyy-mm-dd>-kappa-103-FIRE-<values-of-dist|fmax|symprec>.gz`).
- [X] I included the urls to the Figshare files in the YAML metadata file (`models/<arch_name>/<model_variant>.yml`). If not using Figshare I have included the urls to the cloud storage service in the description of the PR.
- [x] I did include the test script (`test_<arch_name>_<task>.py` for `task` in `discovery`, `kappa`, `diatomics`) that generated the prediction files.

Due to our company’s security policy, it is difficult for us to share the code publicly. In this case, would it still be necessary to upload the test script even if it would not be meaningful without the code?

Thank you for your guidance.

- [ ] I included a training script (`train_<arch_name>.py`) if I trained a model specifically for this benchmark.
- [ ] I included a `readme.md` with additional details about my model.

### CODE REVIEW

**Comment by choant2003** [comment]
> Can you help us understand what this submission is presenting? If the model is just a clone of SevenNet architecture given that we do not yet have good metrics for inference speed it's not clear to me how this fits on the leaderboard. Is the aim of the submission just to show equivalence between this model trained with the FlashTP and sevennet-ompa trained with e3nn?

The main advantage of FlashTP is that it accelerates tensor product operations by 3 to 5 times compared to e3nn, while also significantly improving memory efficiency. Leveraging these benefits, we scaled up the model beyond the original SevenNet architecture. As a result, our model achieved an F1 score of 0.919, K_SRME of 0.158, and RMSD of 0.060, leading to a CPS score of 0.888 — comparable to the current SOTA model, eSEN.

We believe these results are meaningful because they demonstrate how far the NequIP/SevenNet-style architecture can be pushed through computational optimization and model scaling. Based on this, we felt it was valuable to submit our model to the benchmark to contribute to understanding the upper limits of this architecture.

**Comment by janosh** [comment]
> As a result, our model achieved an F1 score of 0.919, K_SRME of 0.158, and RMSD of 0.060, leading to a CPS score of 0.888 — comparable to the current SOTA model, eSEN.

thanks @choant2003, that sounds like a useful data point for the leaderboard!

> ## Additional Information (Optional)
> * [x]  I included a training script (`train_<arch_name>.py`) if I trained a model specifically for this benchmark.
> * [x]  I included a `readme.md` with additional details about my model.

some of the checkmarks in your PR description are currently incorrect. e.g. there's no readme or training script in this PR. not required but always welcome.

> Due to our company’s security policy, it is difficult for us to share the code publicly. In this case, would it still be necessary to upload the test script even if it would not be meaningful without the code?

yes, `test_<arch_name>_<task>.py` for task in `discovery`, `kappa`, `diatomics` would be welcome even without the model weights and code. they are not mandatory for submission but strongly encouraged to enable others to reproduce and benefit from your work

**Comment by choant2003** [comment]
> are you referring to all 3 (model training scripts, model architecture code and the test scripts)? i.e. without any code?

Yes. all three of them.

**Comment by janosh** [comment]
@choant2003 thanks for your patience. i added some more missing metadata in ebc87a6. could you check whether all of that is correct? any update on whether you're able to supply the test scripts?

**Comment by choant2003** [comment]
> The checklist is really the minimal thing we want to except because seeing the test scripts is important because the performance also can depend on things like the choice of optimizer etc.
> 
> GNoME is more a historical artifact that a precedent. When MatterSim was first added before it was later made open source we insisted on all the required content being provided recognizing the mistake we made with GNoME - #104.

Thanks for your reply and clarifications.

I just want to confirm one more point to be extra cautious: if we only provide the test scripts (without model code and checkpoint), is that still acceptable? I understand that in this case the results would not be fully reproducible, but I want to be sure this aligns with the minimum expectations for submission.

In parallel, we’ll start our internal process to open the test scripts code.

**Comment by janosh** [comment]
thanks @choant2003! could you fix the lints listed in [this CI run](https://github.com/janosh/matbench-discovery/actions/runs/18675369660/job/53244130535) and ping me again once lint CI is green?

File: models/equflash/thermal_conductivity.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+            )
+            atoms.calc = calculator
+            forces += [atoms.get_forces()]
+
+    force_set = np.array(forces)
+    ph3.forces = force_set
+    return force_set
+
+
+def init_phono3py(
```
Comment (janosh) [inline models/equflash/thermal_conductivity.py]: @choant2003 what's the reason for duplicating this function rather than importing it from `matbench_discovery/phonons/thermal_conductivity.py`?

https://github.com/choant2003/matbench-discovery/blob/27e2de0dd370b0fd467c05f39fedd5b3b453a87c/matbench_discovery/phonons/thermal_conductivity.py#L116

generally could you try to minimize the amount of duplicate code in this PR?

Comment (choant2003) [reply models/equflash/thermal_conductivity.py]: > @choant2003 what's the reason for duplicating this function rather than importing it from `matbench_discovery/phonons/thermal_conductivity.py`?
> 
> https://github.com/choant2003/matbench-discovery/blob/27e2de0dd370b0fd467c05f39fedd5b3b453a87c/matbench_discovery/phonons/thermal_conductivity.py#L116
> 
> generally could you try to minimize the amount of duplicate code in this PR?

You're right — that's my mistake. I'll clean it up to remove the duplication as much as possible. I'll let you know after I make changes

**Comment by choant2003** [comment]
@janosh 
I’ve refactored the code according to your suggestions and removed some duplicated logic.

For explanation about optimizer code, 
We implemented batchwise relaxation with fretechet filter and FIRE optimizer based on the unitcell filter and batchwise LBFGS optimizer from fairchem-core==1.10.0.

During development, we found that even if the model runs in fp32, the relaxation algorithm itself must operate in fp64 for stable convergence.
So, the model forward is computed in fp32, but the relaxation loop runs in fp64 internally.
For this reason, we didn’t use the fairchem-core optimizable base class and instead implemented it directly.

Thermal conductivity was also modified to compute the 3rd-order force constants batchwise.

Although each evaluation step still has room for optimization — and recent versions of torch-sim have added batchwise relaxation support — we believe this implementation could still be useful for the broader MLFF/MLIP community.
